From 12e6c6ae14c604fae98092e2941a044aac6025eb Mon Sep 17 00:00:00 2001
From: David Sernelius <david@tinygiant.se>
Date: Mon, 11 May 2020 16:22:16 +0200
Subject: [PATCH] mxc_viu.c: removes unrelated pixformat check

* Removes pixformat check in queue_setup. Not related to number of
  buffers or buffer sizes, so not relevant here. Also checks what format is
  already chosen and not what the buffer will be used for which is
  totally irrelevant. Breaks ioctl CREATE_BUFFERS in certain
  circumstances.
---
 drivers/media/platform/mxc/capture/mxc_viu.c | 20 ++------------------
 1 file changed, 2 insertions(+), 18 deletions(-)

diff --git a/drivers/media/platform/mxc/capture/mxc_viu.c b/drivers/media/platform/mxc/capture/mxc_viu.c
index 1ea193853b36..3063470dc041 100644
--- a/drivers/media/platform/mxc/capture/mxc_viu.c
+++ b/drivers/media/platform/mxc/capture/mxc_viu.c
@@ -961,30 +961,14 @@ static int imx_viu_queue_setup(struct vb2_queue *q,
 	struct v4l2_pix_format *fmt = &viu_dev->v4l2_pix_fmt;
 
 	/* TODO: don't support multiple plane format */
+	if (!*num_planes || *num_planes > 1)
+		*num_planes = 1;
 
 	WARN_ON(*num_buffers < IMX_VIU_MIN_BUFS);
 
 	if (sizes[0] <= 0)
 		sizes[0] = fmt->sizeimage;
 
-	switch (fmt->pixelformat) {
-	case V4L2_PIX_FMT_UYVY:
-		if (!*num_planes || *num_planes > 1)
-			*num_planes = 1;
-		break;
-	case V4L2_PIX_FMT_YUYV:
-		if (!*num_planes || *num_planes > 1)
-			*num_planes = 1;
-		break;
-	case V4L2_PIX_FMT_ABGR32:
-		if (!*num_planes || *num_planes > 1)
-			*num_planes = 1;
-		break;
-	default:
-		/* unsupported format */
-		return -EINVAL;
-	}
-
 	return 0;
 }
 
-- 
2.26.2

